### PR TITLE
fixing dopar behaviour

### PR DESCRIPTION
**Pull request by Hong-Revo**

### CODE REVIEW

**Comment by HenrikBengtsson** [comment]
I don't know enough about the original intention for the `data` argument, but I know that the [doRNG](https://cran.r-project.org/package=doRNG) package uses it [by passing internal parameters via a list to `data` of `doRNG()`](https://github.com/renozao/doRNG/blob/master/R/doRNG.R#L149-L174).  It's [`registerDoRNG()`](https://github.com/renozao/doRNG/blob/ebf1dbf4b613d7f65322163d43a001cca087cfee/R/doRNG.R#L488-L506) uses:
```r
setDoPar(doRNG, list(seed=seed, once=once, nseed=0, backend=backend), infoDoRNG)
```
That second argument to `foreach::setDoPar()` is `data`.  So, maybe you want to do the same here to at least not introduce yet another standard; remember, people will look into the code and adapt design patterns.  Again, it would be really useful to know a little bit more how `data` should be used (beyond what's in `?foreach::setDoPar`).

Second comment, which might be more concerning, is that `setDoPar()` takes `data` and assigned it to an external environment:
```r
assign("data", data, pos = .foreachGlobals, inherits = FALSE)
```
Now, because of this, anyone who uses `doRNG::registerDoRNG()` or `doRNG::`%dorng%`` instead of `%dopar%` will end up overriding what ever you assign to `data`, because `registerDoRNG()` calls:
```r
	setDoPar(doRNG, list(seed=seed, once=once, nseed=0, backend=backend), infoDoRNG)
```
In other words, make sure to set your updates with:
```r
library(foreach)
options(foreachDoparLocal = TRUE)
registerDoSEQ()
library(doRNG)
registerDoRNG()
a <- 0
y <- foreach(i = 1:3) %dopar% { a <- i }
stopifnot(a == 0)
```
I suspect the injection of doRNG will override your internal `local` setting.

**Comment by Hong-Revo** [comment]
That should be fine; I only hijack `data` in the case where there is no valid parallel backend. As long as `setDoPar` has been called previously, everything should still work as before.

**Comment by HenrikBengtsson** [comment]
Another comment: this PR is toward code that implements the unresolved discussion in Issue #2.  I think it's safer to work toward the foreach 1.4.7 code base which is what's currently on CRAN.  That way this this feature can be released independently.
